### PR TITLE
DAOS-13777 control: Invalid storage information with MD-on-SSD

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

With storage query usage, only take into account NVMe devices which are used to store data.

# Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

# Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by kjacque — COMMENTED** [review]
Seems reasonable, but I'd like to see some unit test coverage that exercises the changes.

**Comment by knard-intel** [comment]
> Seems reasonable, but I'd like to see some unit test coverage that exercises the changes.

- [ ] Add unit tests covering the change

**Review by tanabarr — COMMENTED** [review]
LGTM, notwithstanding the comment about a unit test case added to verify the change

**Review by tanabarr — APPROVED** [review]

File: src/control/lib/control/pool.go [hunk 1]
```diff
@@ -1266,6 +1266,11 @@ func processSCMSpaceStats(log logging.Logger, filterRank filterRankFn, scmNamesp
 func processNVMeSpaceStats(log logging.Logger, filterRank filterRankFn, nvmeControllers storage.NvmeControllers, rankNVMeFreeSpace rankFreeSpaceMap) error {
 	for _, nvmeController := range nvmeControllers {
 		for _, smdDevice := range nvmeController.SmdDevices {
+			if !smdDevice.Roles.IsEmpty() && (smdDevice.Roles.OptionBits&storage.BdevRoleData) == 0 {
```
Comment (kjacque) [inline src/control/lib/control/pool.go:1269]: Good change for ctl_storage_rpc_test.go, but I was hoping to see a test case for this one as well, showing that we now skip the non-data SSDs.

Comment (knard-intel) [reply src/control/lib/control/pool.go:1269]: - [ ] Add unit test for `processNVMeSpaceStats()` function

Comment (knard-intel) [reply src/control/lib/control/pool.go:1269]: - [ ] Add unit tests for processNVMeSpaceStats() function

Comment (knard-intel) [reply src/control/lib/control/pool.go:1269]: Thanks for the remark.
The unit test allowed me to fix a potential issue.
Indeed, the device was not skipped: forgot the continue statement.
However, as the usable size was reset to zero in the control plane it was not detected with the functional tests.
With future update of the control plane, this last action could have not been done any more and a bug would arise.

**Comment by knard-intel** [comment]
Integrated hot patch 6b222beb4fad52049e0d82cf90d8792c7456236b to fix regression with system name test.
https://daosio.atlassian.net/browse/DAOS-13670?focusedCommentId=117286

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

**Review by kjacque — DISMISSED** [review]

**Comment by knard-intel** [comment]
Remove hotfix commit 6b222beb4fad52049e0d82cf90d8792c7456236b as it will be handle with the dedicated JIRA task [DAOS-13886](https://daosio.atlassian.net/browse/DAOS-13886)

**Comment by knard-intel** [comment]
Resubmit to CI with skipping functional tests related to the JIRA ticket [DAOS-13886](https://daosio.atlassian.net/browse/DAOS-13886)

**Comment by knard-intel** [comment]
PR Build [#11](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12485/11) failure:
* [DaosAggregationBasic:test_basic_aggregation](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_aggregation/DaosAggregationBasic/Test_Hardware___Functional_Hardware_Medium___1___aggregation_basic_py_DaosAggregationBasic_test_basic_aggregation_run_container_hosts_ior_client_processes_iorflags_objectclass_oclass_SX_pool_server_config_engines_0_storage_0_1_80f1/): Failure on too small test timeout => should be a CI issue.
* [DaosAggregationBasic:test_basic_aggregation](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_aggregation/DaosAggregationBasic/Test_Hardware___Functional_Hardware_Medium___2___aggregation_basic_py_DaosAggregationBasic_test_basic_aggregation_run_container_hosts_ior_client_processes_iorflags_objectclass_oclass_RP_2GX_pool_server_config_engines_0_storage_0_1_a4f3/): Failure on too small test timeout => should be a CI issue.
* [AggregationChecksum:test_aggregationchecksum](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_aggregation/AggregationChecksum/Test_Hardware___Functional_Hardware_Medium___1___aggregation_checksum_py_AggregationChecksum_test_aggregationchecksum_run_container_hosts_ior_client_processes_objectclass_SX_transfersize_blocksize_2K_pool_server_config_engines_0_storage_0_1_b17c/): Failure on too small test timeout => should be a CI issue.
* [PoolCreateTests:test_create_no_space_loop](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_pool/PoolCreateTests/Test_Hardware___Functional_Hardware_Medium___1___pool_create_py_PoolCreateTests_test_create_no_space_loop_run_hosts_pool_1_pool_2_server_config_engines_0_storage_0_1_timeouts_ec9e/): CI issue [DAOS-13582](https://daosio.atlassian.net/browse/DAOS-13582) under investigation
* [DaosCoreTest:test_daos_extend_simple](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___20___daos_test_suite_py_DaosCoreTest_test_daos_extend_simple_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_a170/): Failure on too small test timeout => should be a CI issue.
* [DaosCoreTest:test_daos_extend_simple](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/11/testReport/FTEST_daos_test/DAOS_Extend_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___20___daos_test_suite_py_DaosCoreTest_test_daos_extend_simple_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_a170/): Same issue as previous. Missing cmocka results of the previous test.

**Review by tanabarr — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Comment by knard-intel** [comment]
The only test failing is a known CI issue:
* [DAOS-13914](https://daosio.atlassian.net/browse/DAOS-13914): issue of the last build [daos_test.DAOS_Rebuild_Simple](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12485/16/testReport/FTEST_daos_test/DAOS_Rebuild_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___REBUILD27__rebuild_with_dfs_in_flight_punch_create/)

**Pull request merged**